firebase/rules/tests/idx.spec.ts

```diff
@@ -0,0 +1,129 @@
+import 'jest';
+import * as matchers from '../matchers';
+import * as targaryen from 'targaryen';
+
+import { SYSTEM_USER } from '@machinelabs/models';
+
+const anonymousUser = targaryen.util.users.anonymous;
+const rules = require('../database.rules.json');
+
+describe('/idx', () => {
+  const currentUser = { uid: '1', email: 'foo@bar.com', provider: 'github' };
+  const systemUser = { uid: SYSTEM_USER };
+
+  beforeEach(() => {
+    jest.addMatchers(matchers);
+
+    targaryen.util.setFirebaseData({
+      idx: {
+        recent_labs: {
+          '1': {}
+        },
+        user_labs: {
+          '1': {}
+        },
+        lab_executions: {
+          '1': {}
+        },
+        lab_visible_executions: {
+          '1': {}
+        },
+        user_executions: {
+          '1': {}
+        },
+        user_visible_executions: {
+          '1': {}
+        },
+        user_visible_labs: {
+          '1': {}
+        }
+      }
+    });
+
+    targaryen.util.setFirebaseRules(rules);
+  });
+
+  it('All users can read into idx recent labs', () => {
+    expect(anonymousUser).canRead('/idx/recent_labs/1');
+    expect(currentUser).canRead('/idx/recent_labs/1');
+    expect(systemUser).canRead('/idx/recent_labs/1');
+  });
```

Comment:
Similar to the comments above: If something is opened up to everyone, no need to verify this.